engine/src/main/java/org/pentaho/di/imp/Import.java

```diff
@@ -0,0 +1,479 @@
+
+    String begin = df.format( start );
+    String end = df.format( stop );
+    log.logMinimal( BaseMessages.getString( PKG, "Import.Log.StartStop", begin, end ) );
+
+    long seconds = ( stop.getTime() - start.getTime() ) / 1000;
+    if ( seconds <= 60 ) {
+      log.logMinimal( BaseMessages.getString( PKG, "Import.Log.ProcessEndAfter", String.valueOf( seconds ) ) );
+    } else if ( seconds <= 60 * 60 ) {
+      int min = (int) ( seconds / 60 );
+      int rem = (int) ( seconds % 60 );
+      log.logMinimal( BaseMessages.getString( PKG, "Import.Log.ProcessEndAfterLong", String.valueOf( min ), String
+        .valueOf( rem ), String.valueOf( seconds ) ) );
+    } else if ( seconds <= 60 * 60 * 24 ) {
+      int rem;
+      int hour = (int) ( seconds / ( 60 * 60 ) );
+      rem = (int) ( seconds % ( 60 * 60 ) );
+      int min = rem / 60;
+      rem = rem % 60;
+      log.logMinimal( BaseMessages.getString(
+        PKG, "Import.Log.ProcessEndAfterLonger", String.valueOf( hour ), String.valueOf( min ), String
+          .valueOf( rem ), String.valueOf( seconds ) ) );
+    } else {
+      int rem;
+      int days = (int) ( seconds / ( 60 * 60 * 24 ) );
+      rem = (int) ( seconds % ( 60 * 60 * 24 ) );
+      int hour = rem / ( 60 * 60 );
+      rem = rem % ( 60 * 60 );
+      int min = rem / 60;
+      rem = rem % 60;
+      log.logMinimal( BaseMessages.getString(
+        PKG, "Import.Log.ProcessEndAfterLongest", String.valueOf( days ), String.valueOf( hour ), String
+          .valueOf( min ), String.valueOf( rem ), String.valueOf( seconds ) ) );
+    }
+
+    exitJVM( returnCode );
+
+  }
+
+  private static List<String> pickupCmdArguments( String[] a ) {
+    List<String> args = new ArrayList<String>( a.length );
+    for ( String arg : a ) {
+      if ( !arg.isEmpty() ) {
+        args.add( arg );
+      }
+    }
+    return args;
+  }
+
+  private static CommandLineOption createOption( String option, String i18nKey, StringBuilder sb ) {
+    return new CommandLineOption( option, BaseMessages.getString( PKG, i18nKey ), sb );
+  }
+
+  private static CommandLineOption createOption( String option, String i18nKey, StringBuilder sb, boolean yesNo, boolean hidden  ) {
+    return new CommandLineOption( option, BaseMessages.getString( PKG, i18nKey ), sb, yesNo, hidden );
+  }
+
+  /**
+   * Parse an argument as an integer.
+   *
+   * @param option
+   *          Command Line Option to parse argument of
+   * @param def
+   *          Default if the argument is not set
+   * @return The parsed argument or the default if the argument was not specified
+   * @throws KettleException
+   *           Error parsing provided argument as an integer
+   */
+  protected static int parseIntArgument( final CommandLineOption option, final int def ) throws KettleException {
+    if ( !Utils.isEmpty( option.getArgument() ) ) {
+      try {
+        return Integer.parseInt( option.getArgument().toString() );
+      } catch ( NumberFormatException ex ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)